session.go

```diff
@@ -418,11 +422,23 @@ func (s *session) WriteBytes(pkg []byte) (int, error) {
 		return 0, ErrSessionClosed
 	}
 
-	lg, err := s.Connection.send(pkg)
+	s.wlock.Lock()
+	defer s.wlock.Unlock()
+	totalSize, writeSize := len(pkg), 0
+	for totalSize >= maxPacketLen {
+		_, err := s.Connection.send(pkg[writeSize:(writeSize + maxPacketLen)])
+		if err != nil {
+			return writeSize, perrors.Wrapf(err, "s.Connection.Write(pkg len:%d)", len(pkg))
+		}
+		totalSize -= maxPacketLen
+		writeSize += maxPacketLen
+	}
+	lg, err := s.Connection.send(pkg[writeSize:])
```

Comment:
does it need to send if the `totalSize = 0`?

Reply:
IMO, the packet needn't to be send. Addressed.

Reply:
@Lvnszn yes, should return if totalSize already decreased to 0

Reply:
Addressed.